freqtrade/exchange/__init__.py

```diff
@@ -333,6 +333,25 @@ def sell(self, pair: str, ordertype: str, amount: float, rate: float) -> Dict:
         except ccxt.BaseError as e:
             raise OperationalException(e)
 
+    def stoploss_limit(self, pair: str, amount: float, stop_price: float, rate: float) -> Dict:
+        # Only binance is supported
+        if not self.name == 'Binance':
+            raise NotImplementedError(
+                'Stoploss limit orders are implemented only for binance as of now.')
+
+        # Set the precision for amount and price(rate) as accepted by the exchange
+        amount = self.symbol_amount_prec(pair, amount)
+        rate = self.symbol_price_prec(pair, rate)
+        stop_price = self.symbol_price_prec(pair, stop_price)
+
+        # Ensure rate is less than stop price
+        if stop_price <= rate:
+            raise OperationalException(
+                'In stoploss limit order, stop price should be more than limit price')
+
+        return self._api.create_order(pair, 'stop_loss', 'sell',
```

Comment:
i think you'll need to add handling for dry-run here - this will fail as the API does not have a key in this case.
Also, i think you'll need similar exception hanlders than for `sell`...

Reply:
added dry-run for it. however I don't get what you mean by handler for sell. execute_sell cancels the stoploss on exchange via cancel_order function (which does already consider dry-run), and I have already tests for being sure that cancel_order(stoploss_order_id) is called. I don't think it can be related to dry-run this part. let me know if I am missing something.

Reply:
for sell creation we have the following exception handlers, making sure creating a sell in one currency does not crash the bot... failing non-gracefully is never good even if you see it as a critical error.

```
        except ccxt.InsufficientFunds as e:
            raise DependencyException(
                f'Insufficient funds to create limit sell order on market {pair}.'
                f'Tried to sell amount {amount} at rate {rate} (total {rate*amount}).'
                f'Message: {e}')
        except ccxt.InvalidOrder as e:
            raise DependencyException(
                f'Could not create limit sell order on market {pair}.'
                f'Tried to sell amount {amount} at rate {rate} (total {rate*amount}).'
                f'Message: {e}')
        except (ccxt.NetworkError, ccxt.ExchangeError) as e:
            raise TemporaryError(
                f'Could not place sell order due to {e.__class__.__name__}. Message: {e}')
        except ccxt.BaseError as e:
            raise OperationalException(e)
```